### ISSUE TITLE
Possibly-incorrect sensitivity used in forecasts

### ISSUE THREAD

**Issue by dburren** [openaps/oref0#1445]
The code (as at 0.7.1) takes the profile sensitivity (for *now*) and scales that by sensitivityRatio.
It then uses that in making forecasts over the coming hours.

However the profile's declared sensitivity can change over time.

If the sensitivity *does* actually change over time, should we not be using the declared sensitivity for each time block (again scaled by sensitivityRatio)?

A follow-on issue may be that the declared CSF (ratio of ISF:ICR) can change, which affects remainingCarbs and remainingCIpeak.

**Comment by scottleibrand** [comment]
OpenAPS mostly assumes that ISF doesn't change (much) over the course of the day, and models diurnal patterns onto basals. Lots of people asserted based on anecdata that *their* ISF or (usually) CR changes over the day, and we came up with a design for autotuning CR, but no one ever got interested enough to actually implement/test that. Then AndroidAPS implemented fully tunable profiles, and lots of people went off and implemented whatever they thought was correct for them, and everyone lost interest in coming up with a more general solution.

In short, we don't have any data to indicate whether the behavior you describe would work better than the current behavior, and testing it would be quite difficult given how variable the source of the original profile sensitivity schedule is.

But if you'd like to, we could probably figure out how to implement a change and test it in the oref0-backtest simulator.
